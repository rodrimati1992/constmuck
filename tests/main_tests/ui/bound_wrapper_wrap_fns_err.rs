use bytemuck::TransparentWrapper;

struct Foo<T: ?Sized>(T);






fn _test_wrap<T>(a: T, b: T) -> (Foo<T>, Foo<T>) {
    (
        constmuck::wrapper::wrap::<Foo<T>, T>(a),
        <Foo<T> as TransparentWrapper<T>>::wrap(b),
    )
}




fn _test_wrap_ref<'a, T>(a: &'a T, b: &'a T, c: &'a T) -> (&'a Foo<T>, &'a Foo<T>, &'a Foo<T>) {
    (
        constmuck::wrapper::wrap_ref::<Foo<T>, T>(a),
        constmuck::wrapper::wrap_ref!(b, Foo<T>, T),
        <Foo<T> as TransparentWrapper<T>>::wrap_ref(c),
    )
}



fn _test_wrap_slice<'a, T>(a: &'a [T], b: &'a [T]) -> (&'a [Foo<T>], &'a [Foo<T>]) {
    (
        constmuck::wrapper::wrap_slice::<Foo<T>, T>(a),
        <Foo<T> as TransparentWrapper<T>>::wrap_slice(b),
    )
}



fn main(){}