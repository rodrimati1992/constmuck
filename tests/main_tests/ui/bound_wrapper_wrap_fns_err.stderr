error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:12:36
   |
12 |         constmuck::wrapper::wrap::<Foo<T>, T>(a),
   |                                    ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::wrap`
  --> src/wrapper.rs:31:12
   |
29 | pub const fn wrap<Outer, Inner>(val: Inner) -> Outer
   |              ---- required by a bound in this function
30 | where
31 |     Outer: TransparentWrapper<Inner>,
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `wrap`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:13:10
   |
13 |         <Foo<T> as TransparentWrapper<T>>::wrap(b),
   |          ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:22:40
   |
22 |         constmuck::wrapper::wrap_ref::<Foo<T>, T>(a),
   |                                        ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::wrap_ref`
  --> src/wrapper.rs:65:12
   |
63 | pub const fn wrap_ref<Outer, Inner>(reff: &Inner) -> &Outer
   |              -------- required by a bound in this function
64 | where
65 |     Outer: TransparentWrapper<Inner>,
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `wrap_ref`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
   --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:23:9
    |
23  |         constmuck::wrapper::wrap_ref!(b, Foo<T>, T),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
    |
    = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck_internal::__WrapRefArgs::<'a, Outer, Inner>::__call`
   --> constmuck_internal/src/lib.rs:103:16
    |
101 |     pub const unsafe fn __call(self) -> &'a Outer
    |                         ------ required by a bound in this associated function
102 |     where
103 |         Outer: TransparentWrapper<Inner>
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `__WrapRefArgs::<'a, Outer, Inner>::__call`
    = note: this error originates in the macro `$crate::wrapper_inner` which comes from the expansion of the macro `constmuck::wrapper::wrap_ref` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:24:10
   |
24 |         <Foo<T> as TransparentWrapper<T>>::wrap_ref(c),
   |          ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
   --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:32:42
    |
32  |         constmuck::wrapper::wrap_slice::<Foo<T>, T>(a),
    |                                          ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
    |
    = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
note: required by a bound in `constmuck::wrapper::wrap_slice`
   --> src/wrapper.rs:139:12
    |
137 | pub const fn wrap_slice<Outer, Inner>(reff: &[Inner]) -> &[Outer]
    |              ---------- required by a bound in this function
138 | where
139 |     Outer: TransparentWrapper<Inner>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `wrap_slice`

error[E0277]: the trait bound `Foo<T>: TransparentWrapper<T>` is not satisfied
  --> tests/main_tests/ui/bound_wrapper_wrap_fns_err.rs:33:10
   |
33 |         <Foo<T> as TransparentWrapper<T>>::wrap_slice(b),
   |          ^^^^^^ the trait `TransparentWrapper<T>` is not implemented for `Foo<T>`
   |
   = help: the trait `TransparentWrapper<T>` is implemented for `Wrapping<T>`
